### ISSUE TITLE
Download/Snapshot Image tool: bugs and improvements

### ISSUE THREAD

**Issue by mirnasilva** [OHIF/Viewers#1217]
### Issue 1: Minimum width and height not defined

1. When user set the width or height to 0 and click on Download button, it doesn't open the download dialog and also doesn't show any error message. 
**Expected Behavior:** The application should display an error message to inform the user that the minimum valid size is 100px, and user should not be able to download the image.

2. When user set the width or height to less than 100px and click on Download button, the downloaded image has size 100x100. Can we set the minimum value to 100px and restrict the user to set values smaller than that? 
**Expected Behavior:**  The application should display an error message to inform the user that the minimum valid size is 100px, and user should not be able to download the image.

### Issue 2: Image is being resized while user still typing the dimensions (width or height)

When user adds a bigger value on height/width field (for example: `10000`), the added value takes some time to be displayed/updated. This is providing a bad user experience because it is updating the image preview at the same type user still typing the value.
**Suggestion:** The Image Preview should be updated only after the user removes the focus of the height/width field. This way we can avoid too many unnecessary on-demand updates on the Image Preview.

### Issue 3: Empty file name

When user deletes the file name no error message is being displayed. Although during the download, the file is displayed with the default name `image.jpg`, letting the user know that the file name is a mandatory field would improve the usability.

### Improvement 4: Width and Height are scaled proportionally

To improve user experience it would be nice to have a visual indication that the image size is being scaled proportionally, like the image below. This way we can avoid having the user trying to set different dimensions unsuccessfully. 

<img width="238" alt="Screen Shot 2019-11-21 at 21 16 38" src="https://user-images.githubusercontent.com/13245529/69445588-65298580-0d31-11ea-802c-74f142d00b16.png">
